### PR TITLE
Feat(web-react): Introduce RouterProvider for client-side routing

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description

- Introduces an opt-in `RouterProvider` for client-side routing in `@alma-oss/spirit-web-react`.
- Consolidates router-aware click handling through `handleLinkClick` and a reusable `useLinkClick` hook.
- Applies the shared behavior across link-like components (`Link`, `ButtonLink`, `Pagination`, `Header`, `Tabs`) with support for `routerOptions`.
- Aligns navigation behavior with **React Aria-inspired guarding rules** to preserve native browser behavior.
- Updates docs
- Expands test coverage for routing edge cases and component integration.

### Code example
```tsx
<RouterProvider navigate={router.push}>
  <Link href="/jobs" routerOptions={{ scroll: false }}>Jobs</Link>
</RouterProvider>
```

### Additional context
https://github.com/adobe/react-spectrum/blob/main/packages/react-aria/src/utils/openLink.tsx
https://github.com/adobe/react-spectrum/blob/main/packages/react-aria/src/link/useLink.ts
https://github.com/adobe/react-spectrum/blob/main/packages/dev/docs/pages/react-aria/routing.mdx

### Issue reference

[DS-2346](https://jira.almacareer.tech/browse/DS-2346)

### CODE REVIEW

File: packages/web-react/src/components/Link/Link.tsx [hunk 1]
```diff
@@ -25,12 +26,22 @@ const _Link = <E extends ElementType = 'a', T = void>(
     children,
     ...restProps
   } = propsWithDefaults;
+  const { href, target, isDisabled, onClick } = restProps;
+  const router = useRouterContext();
   const { classProps, props: modifiedProps } = useLinkStyleProps(restProps);
   const { styleProps, props: otherProps } = useStyleProps(modifiedProps);
   const mergedStyleProps = mergeStyleProps(ElementTag, { classProps, styleProps, otherProps });
 
+  const handleClick = getRouterClickHandler({
```
Comment (literat) [inline packages/web-react/src/components/Link/Link.tsx]: suggestion(if-minor): 
```
handleLinkClick(e, router, item!.props.href, item?.props.routerOptions);
```

Comment (literat) [reply packages/web-react/src/components/Link/Link.tsx]:
```jsx
import { handleLinkClick } from ...

const handleClick = (...) => {
  handleLinkClick(...)
}
```

File: packages/web-react/src/components/Pagination/PaginationButtonLink.tsx [hunk 2]
```diff
@@ -15,14 +17,26 @@ const _PaginationButtonLink = <E extends ElementType = 'a'>(
...
+    href,
+    target,
+    download,
+    onClick,
+    isDisabled,
+  });
+
   return (
-    <ButtonLink color="secondary" isSymmetrical {...restProps} ref={ref}>
+    <ButtonLink color="secondary" isSymmetrical {...restProps} href={href} onClick={handleClick} ref={ref}>
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web-react/src/components/Pagination/PaginationButtonLink.tsx]: suggestion (non-blocking): PaginationButtonLink builds a handleClick handler and passes it to ButtonLink, which then builds a second handleLinkClick handler wrapping the first

Either replace `ButtonLink` with a plain <Component> (as all other components do), or skip the `handleLinkClick` call in PaginationButtonLink and rely on ButtonLink's own integration. Since ButtonLink already     reads the same router context and handles the same props, the wrapper is redundant here.

When `shouldUseRouter` is `true` in `PaginationButtonLink`, the returned handler becomes the `onClick` passed into `ButtonLink`. `ButtonLink` receives a router context, sees `href` is internal, and wraps that handler again. The double wrapping doesn't break anything today (the inner ButtonLink handler calls the outer one, which navigates), but it makes the call chain difficult to reason about and the `preventDefault()` is called twice.

File: packages/web-react/src/utils/router.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ * @param {string} [options.target] HTML link target.
+ * @param {string | boolean} [options.download] HTML link download attribute.
+ * @param {MouseEventHandler} [options.onClick] Original click handler.
+ * @param {boolean} [options.isDisabled] Whether the link is disabled.
+ */
+export const handleLinkClick = (options: HandleLinkClickOptions): MouseEventHandler | undefined => {
+  const { router, routerOptions, href, target, download, onClick, isDisabled } = options;
+  const isNonPathHref = typeof href === 'string' && isExternalOrNonPathHref(href);
+  const shouldUseRouter = Boolean(
+    router && href && !isDisabled && !download && !shouldDeferToBrowserForTarget(target) && !isNonPathHref,
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/utils/router.ts]: should be fixed

File: packages/web-react/src/utils/router.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  );
+
+  if (!shouldUseRouter) {
+    return onClick;
+  }
+
+  return (event) => {
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/utils/router.ts]: Intentional for this implementation.
We currently early-return when event.defaultPrevented is already set before handler execution (no onClick, no navigate), and this behavior is now documented in JSDoc to avoid ambiguity.
If we want to change this to always call consumer onClick even when already prevented, that would be a behavioral decision we can discuss in a follow-up PR.

File: packages/web-react/src/hooks/useLinkClick.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+'use client';
+
+import { type MouseEventHandler } from 'react';
+import { useRouter } from '../context/RouterContext';
+import { handleLinkClick } from '../utils/router';
+
+type UseLinkClickOptions = {
```
Comment (literat) [inline packages/web-react/src/hooks/useLinkClick.ts]: suggestion: This type is very similar to `HandleLinkClickOptions`. I think one should be the base type and the other the extension.

File: packages/web-react/src/utils/__tests__/router.test.ts [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+  it('should skip router for empty-string download attribute', () => {
+    const navigate = jest.fn();
+    const onClick = jest.fn();
+    const handler = handleLinkClick({ router: { navigate }, href: '/test', download: '', onClick });
+
+    expect(handler).toBe(onClick);
+  });
+
+  it('should not navigate on modified or non-primary clicks', () => {
```
Comment (literat) [inline packages/web-react/src/utils/__tests__/router.test.ts]: Missing test: `onRouterClick` with modified click still calls onClick

todo (non-blocking): The test "should not navigate on modified or non-primary
clicks" verifies navigate is not called and onClick is called — but the assertion
for onClick is not present in the modifier branch

Fix: Add `expect(onClick).toHaveBeenCalledTimes(1)` after
`expect(navigate).not.toHaveBeenCalled()` for both the `metaKey` and non-primary-button
branches to explicitly document and enforce that `onClick` is still called on modifier
clicks.

Why: The `onRouterClick` handler does call onClick on modifier clicks
(shouldDeferToBrowserForLinkClick branch, line 76–78), but this behaviour is not
asserted in the tests. A future refactor that accidentally drops the onClick
invocation in that branch would go undetected.

File: examples/next-with-pages-router/tsconfig.tsbuildinfo
Comment (literat) [inline examples/next-with-pages-router/tsconfig.tsbuildinfo:1]: issue: Please, remove this file.

File: packages/web-react/src/components/ButtonLink/useButtonLinkStyleProps.ts [hunk 7]
```diff
@@ -12,8 +12,9 @@ const getButtonLinkColorClassname = <C = void>(className: string, color: ButtonC
 const getButtonLinkSizeClassname = <S = void>(className: string, size: ButtonSize<S>): string =>
   compose(applySize<ButtonSize<S>>(size))(className);
 
-export function useButtonLinkStyleProps<C = void, S = void>(props: ButtonStyleProps<C, S>) {
-  const { color, isBlock, isDisabled, isLoading, isSymmetrical, size, ...restProps } = props;
+export function useButtonLinkStyleProps<C = void, S = void>(props: ButtonLinkStyleProps<C, S>) {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars -- strip `routerOptions` so it is not forwarded to the DOM
+  const { color, isBlock, isDisabled, isLoading, isSymmetrical, size, routerOptions, ...restProps } = props;
```
Comment (literat) [inline packages/web-react/src/components/ButtonLink/useButtonLinkStyleProps.ts]: issue: I think this is not a proper place to remove the `routerOptions`.

File: packages/web-react/src/components/Link/useLinkStyleProps.ts [hunk 8]
```diff
@@ -3,7 +3,8 @@ import { useClassNamePrefix } from '../../hooks';
 import { type LinkProps, type LinkStyleProps, UNDERLINED_OPTIONS } from '../../types';
 
 export function useLinkStyleProps<C = void>(props: LinkStyleProps<C>) {
-  const { color, hasVisitedStyleAllowed, isDisabled, underlined, ...restProps } = props;
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars -- strip `routerOptions` so it is not forwarded to the DOM
+  const { color, hasVisitedStyleAllowed, isDisabled, underlined, routerOptions, ...restProps } = props;
```
Comment (literat) [inline packages/web-react/src/components/Link/useLinkStyleProps.ts]: issue: This is not a proper place to remove the `routerOptions`.

File: packages/web-react/src/utils/router.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    router &&
+    href &&
+    !isDisabled &&
+    !hasDownloadAttribute &&
+    !shouldDeferToBrowserForTarget(target) &&
+    !isNonPathHref &&
+    !isHashHref,
+  );
+
+  const onNativeClick = onClick;
```
Comment (literat) [inline packages/web-react/src/utils/router.ts]: suggestion: Remove this line.

File: packages/web-react/src/utils/router.ts [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+ * @param {RouterContextType} options.router Router context used for navigation.
+ * @param {Record<string, unknown>} [options.routerOptions] Extra options for router navigation.
+ * @param {string} [options.href] Link target URL.
+ * @param {string} [options.target] HTML link target.
+ * @param {string | boolean} [options.download] HTML link download attribute.
+ * @param {MouseEventHandler} [options.onClick] Original click handler.
+ * @param {boolean} [options.isDisabled] Whether the link is disabled.
+ */
+export const handleLinkClick = (options: HandleLinkClickOptions): MouseEventHandler | undefined => {
+  const { router, routerOptions, href, target, download, onClick, isDisabled } = options;
```
Comment (literat) [inline packages/web-react/src/utils/router.ts]:
```suggestion
  const { router, routerOptions, href, target, download, onClick: onNativeClick, isDisabled } = options;
```

**Comment by literat** [comment]
> DELETE DEMO FILES BEFORE MERGE

What about keeping them in a page like `/routing`?

File: packages/web-react/src/components/Pagination/PaginationLink.tsx [hunk 11]
```diff
@@ -1,25 +1,35 @@
...
 import { mergeStyleProps } from '../../utils';
 import { VisuallyHidden } from '../VisuallyHidden';
 import { usePaginationStyleProps } from './usePaginationStyleProps';
 
 const _PaginationLink = <E extends ElementType = 'a'>(props: SpiritPaginationLinkProps<E>, ref: PolymorphicRef<E>) => {
   const { t } = useI18n();
-  const { elementType = 'a', accessibilityLabel, isCurrent, pageNumber, ...restProps } = props;
+  const { elementType = 'a', accessibilityLabel, isCurrent, pageNumber, routerOptions, ...restProps } = props;
   const visuallyHiddenLabel = accessibilityLabel || `${t('pagination.goToPage')} ${pageNumber}`;
+  const { href } = restProps;
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/Pagination/PaginationLink.tsx]: yes it is. removed

File: packages/web-react/src/hooks/useLinkClick.ts (hunk 5 above)
Comment (pavelklibani) [reply packages/web-react/src/hooks/useLinkClick.ts]: changed

File: packages/web-react/src/utils/__tests__/router.test.ts (hunk 6 above)
Comment (pavelklibani) [reply packages/web-react/src/utils/__tests__/router.test.ts:135]: did as you recommended 👍

File: examples/next-with-pages-router/src/pages/RouterDemo.tsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
+import { Link, RouterProvider } from '@alma-oss/spirit-web-react';
+import { useRouter, type NextRouter } from 'next/router';
+
+type NextPagesRouterOptions = NonNullable<Parameters<NextRouter['push']>[2]>;
+
+const RouterDemo = () => {
+  const router = useRouter();
+  const navigate = (path: string, opts?: NextPagesRouterOptions) => router.push(path, undefined, opts);
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply examples/next-with-pages-router/src/pages/RouterDemo.tsx]: done

File: examples/next-with-pages-router/tsconfig.tsbuildinfo
Comment (pavelklibani) [reply examples/next-with-pages-router/tsconfig.tsbuildinfo:1]: removed

File: packages/web-react/src/components/ButtonLink/useButtonLinkStyleProps.ts (hunk 7 above)
Comment (pavelklibani) [reply packages/web-react/src/components/ButtonLink/useButtonLinkStyleProps.ts]: removing routerOptions outside this hook 👍

File: packages/web-react/src/utils/router.ts (hunk 9 above)
Comment (pavelklibani) [reply packages/web-react/src/utils/router.ts]: removed

File: packages/web-react/src/utils/router.ts (hunk 10 above)
Comment (pavelklibani) [reply packages/web-react/src/utils/router.ts]: done

File: packages/web-react/src/components/ButtonLink/useButtonLinkStyleProps.ts [hunk 13]
```diff
@@ -12,8 +12,9 @@ const getButtonLinkColorClassname = <C = void>(className: string, color: ButtonC
 const getButtonLinkSizeClassname = <S = void>(className: string, size: ButtonSize<S>): string =>
   compose(applySize<ButtonSize<S>>(size))(className);
 
-export function useButtonLinkStyleProps<C = void, S = void>(props: ButtonStyleProps<C, S>) {
-  const { color, isBlock, isDisabled, isLoading, isSymmetrical, size, ...restProps } = props;
+export function useButtonLinkStyleProps<C = void, S = void>(props: ButtonLinkStyleProps<C, S>) {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/ButtonLink/useButtonLinkStyleProps.ts]: this change is outdated, no longer there

File: packages/web-react/src/components/Link/useLinkStyleProps.ts [hunk 14]
```diff
@@ -3,7 +3,8 @@ import { useClassNamePrefix } from '../../hooks';
 import { type LinkProps, type LinkStyleProps, UNDERLINED_OPTIONS } from '../../types';
 
 export function useLinkStyleProps<C = void>(props: LinkStyleProps<C>) {
-  const { color, hasVisitedStyleAllowed, isDisabled, underlined, ...restProps } = props;
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/Link/useLinkStyleProps.ts]: this change is outdated, no longer there

File: packages/web-react/src/context/RouterContext.ts [hunk 15]
```diff
@@ -0,0 +1,42 @@
+'use client';
+
+import { type ReactNode, createContext, createElement, useContext, useMemo } from 'react';
+
+type RouterContextType = {
+  navigate(path: string, routerOptions?: Record<string, unknown>): void | Promise<void>;
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/context/RouterContext.ts]: done

File: packages/web-react/src/utils/router.ts [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+const shouldDeferToBrowserForTarget = (target?: string): boolean =>
+  typeof target === 'string' && target.toLowerCase() !== '_self';
+
+const shouldDeferToBrowserForLinkClick = (
+  event: Pick<MouseEvent, 'metaKey' | 'ctrlKey' | 'shiftKey' | 'altKey' | 'button'>,
+): boolean => {
+  if (event.metaKey || event.ctrlKey || event.shiftKey || event.altKey) {
+    return true;
+  }
+  if (typeof event.button === 'number' && event.button !== 0) {
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/utils/router.ts]: empty lines added

File: packages/web-react/src/utils/router.ts [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+    !isHashHref,
+  );
+
+  const onNativeClick = onClick;
+
+  const onRouterClick: MouseEventHandler = (event) => {
+    if (event.defaultPrevented) {
+      return;
+    }
+    if (shouldDeferToBrowserForLinkClick(event)) {
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/utils/router.ts]: added

File: packages/web-react/src/utils/router.ts [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+    !isDisabled &&
+    !hasDownloadAttribute &&
+    !shouldDeferToBrowserForTarget(target) &&
+    !isNonPathHref &&
+    !isHashHref,
+  );
+
+  const onNativeClick = onClick;
+
+  const onRouterClick: MouseEventHandler = (event) => {
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/utils/router.ts:65]: done

File: packages/web-react/src/utils/router.ts [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+import { type RouterContextType } from '../context/RouterContext';
+
+const isProtocolRelativeUrl = (href: string): boolean => href.trim().startsWith('//');
+const hasNonPathUrlScheme = (href: string): boolean => /^[a-z][a-z0-9+.-]*:/i.test(href.trim());
+const isExternalOrNonPathHref = (href: string): boolean => hasNonPathUrlScheme(href) || isProtocolRelativeUrl(href);
+const isHashOnlyHref = (href: string): boolean => href.trim().startsWith('#');
+const shouldDeferToBrowserForTarget = (target?: string): boolean =>
+  typeof target === 'string' && target.toLowerCase() !== '_self';
+
+const shouldDeferToBrowserForLinkClick = (
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/utils/router.ts:11]: done

File: packages/web-react/src/hooks/useLinkClick.ts [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+  routerOptions?: Record<string, unknown>;
+  href?: string;
+  target?: string;
+  download?: string | boolean;
+  onClick?: MouseEventHandler;
+  isDisabled?: boolean;
+};
+
+export const useLinkClick = (options: UseLinkClickOptions): MouseEventHandler | undefined => {
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/hooks/useLinkClick.ts:9]: done

File: packages/web-react/src/components/Header/HeaderDialogLink.tsx [hunk 21]
```diff
@@ -15,7 +15,8 @@ const _HeaderDialogLink = <E extends ElementType = 'a'>(
   props: SpiritDialogHeaderLinkProps<E>,
   ref: PolymorphicRef<E>,
 ): JSX.Element => {
-  const { elementType = 'a', children, isCurrent, ...restProps } = props;
+  const { elementType = 'a', children, isCurrent, routerOptions, ...restProps } = props;
+  const { href } = restProps;
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/Header/HeaderDialogLink.tsx]: removed

**Comment by pavelklibani** [comment]
rebased with main
